compiler/semcall.nim

```diff
@@ -717,8 +726,12 @@ proc searchForBorrowProc(c: PContext, startScope: PScope, fn: PSym): PSym =
     var resolved = semOverloadedCall(c, call, call, filter, {})
     if resolved != nil:
       result = resolved[0].sym
+      while sfBorrow in result.flags:
+        # We iterate over all borrowed procs.
+        # This ensures we always have an instantiated proc to call
+        result = result.ast[bodyPos].sym
```

Comment:
No, the underlying proc was not called here, the borrowed one was. Think about "find all usages" -- you want to list the usages of the borrowed proc, not the usages of the underlying proc.

Reply:
Without that logic the following errors due to Baz's proc pointing to Bar's proc which isn't instantiated. Erroring with `Error: internal error: expr: proc not init $`. Presently transf doesnt handle this, but if it should I can move the logic there.
```nim
type
  Foo = distinct seq[int]
  Bar[N: static[int]] = distinct seq[int]
  Baz = distinct Bar[10]
  
proc `$`(s: Foo): string {.borrow.}
proc `$`(s: Bar): string {.borrow.}
proc `$`(s: Baz): string {.borrow.}
```

Reply:
Yes, patch `transf` instead.